### ISSUE TITLE
[BUG]: UI unresponsive when switching feeds while sync is in progress

### ISSUE THREAD

**Issue by pprkut** [martinrotter/rssguard#1236]
### Brief description of the issue

When I click another feed in the left navigation pane, the UI locks up seemingly until feed synchronization progresses to the next feed. Even the systray is unresponsive in that period. Switching articles in the already selected feed does not have a similar affect.

I'm syncing from nextcloud, but not sure that matters.

### How to reproduce the bug?

1. Wait until feed synchronization starts
2. Switch feeds

### What was the expected result?

The UI should be unaffected by the ongoing sync progress and switching from one feed to another should be (near) instant.

### What actually happened?

The complete UI locks up until synchronization progresses to the next feed

### Debug log

-

### Operating system and version

* OS: Slackware Linux
* RSS Guard version: 4.5.5

**Comment by martinrotter** [comment]
Damn, I know about the issue, it just did not really show up for me, ever.

Would you be somehow able to generate or anonymize your database/config and provide it for testing? I believe this is somehow connected to db size and/or config. Not all user experience this, issue, thats for sure.

**Comment by xmstspider** [comment]
Hmm, not sure, there is some internal stuff, so maybe, but can't promise anything

A quick idea - maybe if you write a SQL test-script to fill the RSS DB directly (because you know the structure) with random articles, like 100k, 200k, ... and check when the problem starts?

**Comment by martinrotter** [comment]
I likely will, nevermind. I will generate the data myself. Thanks.

**Comment by pprkut** [comment]
I tried running a sync process + switching feed a couple times through perf. No debugging symbols though, so not entirely sure it's useful. Looking through the data it might be related to a db locking issue though. It spends a lot of time in `DatabaseQueries::updateMessages`.

[perf.data.perfparser.gz](https://github.com/martinrotter/rssguard/files/13601501/perf.data.perfparser.gz)

**Comment by martinrotter** [comment]
Yes, problem is VEEEERY likely connected to fact you have huge DB file. Will have to create one somehow.

**Comment by martinrotter** [comment]
Closing, re-test with latest stable build and let me know if it persists.
